contracts/utils/cryptography/EIP712.sol

```diff
@@ -71,16 +78,12 @@ abstract contract EIP712 {
         if (address(this) == _CACHED_THIS && block.chainid == _CACHED_CHAIN_ID) {
             return _CACHED_DOMAIN_SEPARATOR;
         } else {
-            return _buildDomainSeparator(_TYPE_HASH, _HASHED_NAME, _HASHED_VERSION);
+            return _buildDomainSeparator();
         }
     }
 
-    function _buildDomainSeparator(
-        bytes32 typeHash,
-        bytes32 nameHash,
-        bytes32 versionHash
```

Comment:
I think this function was defined with arguments initially because immutable variables couldn't be read in the constructor. Is this fixed for Solidity >=0.8.8?

The same immutable constraints where the reason for a few of the other quirks in how this contract was written.

Reply:
Yes it was fixed in 0.8.8, which is the same version where UDVT were introduced.

Since ShortString uses UDVT, we have to bump to 0.8.8, so we can also use the other features provided by that version